docs: Edge Function docs (#18738)
supabase/supabase: apps/docs/pages/guides/functions/regional-invocation.mdx

export const meta = {
  id: 'function-regional-invocation',
  title: 'Regional Invocations',
  description: 'How to execute an Edge Functions in a particular region.',

>   subtitle: 'How to execute an Edge Functions in a particular region.',

Reviewer: ```suggestion
  subtitle: 'How to execute an Edge Function in a particular region.',
```

}

Edge Functions are executed in the region closest to the user making the request. This helps to reduce network latency and provide faster responses to the user.

> However, if your Function performs lot of database or storage operations, invoking the Function in the same region as your database may provide better performance. Some situations where this might be helpful include:

Reviewer: ```suggestion
However, if your Function performs lots of database or storage operations, invoking the Function in the same region as your database may provide better performance. Some situations where this might be helpful include:
```

- Bulk adding and editing records in your database
- Uploading files

Supabase provides an option to specify the region when invoking the Function.

## Using the `x-region` header

Use the `x-region` HTTP header when calling an Edge Function to determine where the Function should be executed:

<CH.Code>

```bash cURL
# https://supabase.com/docs/guides/functions/deploy#invoking-remote-functions
curl --request POST 'https://<project_ref>.supabase.co/functions/v1/hello-world' \
  --header 'Authorization: Bearer ANON_KEY' \
  --header 'Content-Type: application/json' \
  --header 'x-region: eu-west-3' \
  --data '{ "name":"Functions" }'
```

```js JavaScript
// https://supabase.com/docs/reference/javascript/installing
import { createClient } from '@supabase/supabase-js'

// Create a single supabase client for interacting with your database
const supabase = createClient('https://xyzcompany.supabase.co', 'public-anon-key')

// https://supabase.com/docs/reference/javascript/functions-invoke
const { data, error } = await supabase.functions.invoke('hello-world', {
  body: { name: 'Functions' },
  headers: { 'x-region': 'eu-west-3' },
})
```

</CH.Code>

> You can verify the region the function was executed by looking at the `x-sb-edge-region` HTTP header in the response. You can also find it as metadata in [Edge Function Logs](/docs/guides/functions/debugging).

Reviewer: ```suggestion
You can verify the execution region by looking at the `x-sb-edge-region` HTTP header in the response. You can also find it as metadata in [Edge Function Logs](/docs/guides/functions/debugging).
```
